libminifi/src/properties/Properties.cpp

```diff
@@ -75,7 +73,7 @@ const core::PropertyValidator* getValidator(const std::string& lookup_value) {
 std::optional<std::string> ensureTimePeriodValidatedPropertyHasExplicitUnit(const core::PropertyValidator* const validator, std::string& value) {
   if (validator != &core::StandardPropertyTypes::TIME_PERIOD_TYPE)
     return std::nullopt;
-  if (value.empty() || !std::all_of(value.begin(), value.end(), [](unsigned char c){ return ::isdigit(c); }))
+  if (value.empty() || !ranges::all_of(value, [](const auto c){ return ::isdigit(c); }))
```

Comment:
The conversion to `unsigned char` was intentional, because it is a quirk of the standard that the parameter type of `isdigit` is `int`, but its behavior is undefined if the parameter is negative (see https://en.cppreference.com/w/c/string/byte/isdigit).

Reply:
ahh, makes sense
I've reverted to to unsigned chars and added a comment to avoid similar changes in the future.
https://github.com/apache/nifi-minifi-cpp/pull/1825/commits/f94809bd07754f9def422b50a4aeaae6bf0323f1